lib/ConsumerImpl.h

```diff
@@ -302,6 +302,7 @@ class ConsumerImpl : public ConsumerImplBase {
 
     const long expireTimeOfIncompleteChunkedMessageMs_;
     DeadlineTimerPtr checkExpiredChunkedTimer_;
+    bool expireChunkMessageTaskScheduled_ = false;
```

Comment:
You should use an `atomic_bool` here and use CAS to set it true.

I see Java client doesn't use `AtomicBoolean`, but `processMessageChunk` could be called in different threads.

Reply:
Good point.